dataprofiler/tests/data_readers/test_data_utils.py

```diff
@@ -36,3 +37,127 @@ def test_file_UTF_encoding_detection(self):
             detected_encoding = \
                 data_utils.detect_file_encoding(file_path=input_file["path"])
             self.assertEqual(detected_encoding.lower(), input_file["encoding"])
+
+    def test_nth_loc_detection(self):
+        """
+        Tests the ability for the `data_utils.find_nth_location` to detect the
+        nth index of a search_query in a string. 
+        """
+        # Input args: string, query, n
+        # Expected results: index, occurances
+        test_queries = [
+            dict(string="This is a test.", query=".", n=1, index=14, occurances=1),
+            dict(string="This is a test\n", query="\n", n=1, index=14, occurances=1),
+            dict(string="This is a test\nThis is a second test\n",
+                 query="\n", n=0, index=-1, occurances=0),
```

Comment:
Returning an index of -1 seems somewhat misleading here, since this implies the 0th occurrence of the query was at the end of the string, which isn't true as that is the 2nd occurrence (in this case), would it be better to return None instead?